be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/ElectionHandler.scala

```diff
@@ -108,7 +108,7 @@ class ElectionHandler(dbRef: => AskableActorRef) extends MessageHandler {
     }
     val electionChannel: Channel = rpcMessage.getParamsChannel
     val combined = for {
-      electionQuestionResults <- createElectionQuestionResults(electionChannel)
+      electionQuestionResults <- createElectionQuestionResults(electionChannel, rpcMessage.extractLaoId)
```

Comment:
Be careful here. What you are extracting is the `electionId` and not the `laoId`. Indeed, the function `extractLaoId` extracts the last id in the channel defined, i.e. for a channel `root/laoId/electionId` the last substring extracted is `electionId`. If you want to access the `laoId`, you need to use the function `decodeChannelLaoId`. 
We need to be careful with `EndElection` messages because those ones are sent on the **election** channel. 
For the `SetupElection` messages, they are sent on the main channel, i.e the lao channel, which means that the channel is as `root/laoId`. So, there is no problem using the `extractLaoId`, because it will take the last substring, meanin the `laoId` ;)
Also, see protocol (and slack) for more informations.